caffe2/operators/expand_op.h

```diff
@@ -0,0 +1,149 @@
+#ifndef CAFFE2_OPERATORS_REDUCE_OPS_H_
+#define CAFFE2_OPERATORS_REDUCE_OPS_H_
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+
+#include "caffe2/core/context.h"
+#include "caffe2/core/operator.h"
+#include "caffe2/core/types.h"
+#include "caffe2/utils/math.h"
+
+namespace caffe2 {
+
+template <typename InputTypes, class Context, class Expander>
+class ExpandOp final : public Operator<Context> {
+ public:
+  USE_OPERATOR_CONTEXT_FUNCTIONS;
+
+  ExpandOp(const OperatorDef& operator_def, Workspace* ws)
+      : Operator<Context>(operator_def, ws) {}
+
+  bool RunOnDevice() override {
+    return DispatchHelper<InputTypes>::call(this, Input(0));
+  }
+ template <typename T>
+  bool DoRunWithType() {
+    const auto& X = Input(0);
+    const auto& Y_shape_tensor = Input(1);
+	std::vector<int> shape_dims(Y_shape_tensor.size());
+	context_.template Copy<int, Context, CPUContext>(Y_shape_tensor.size(), Y_shape_tensor.template data<int>(), shape_dims.data());
+    const int* Y_shape = Y_shape_tensor.template data<int>();
```

Comment:
nit: move this to one line above so the `Copy` call can use this variable

Reply:
Infact I can just delete this var since it is no use anymore...